docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc

```diff
@@ -1,12 +1,115 @@
+intervening leap seconds, so that the number of minutes and seconds past the 
+hour is the same at the start and end.
+* Multiple minutes (_n_m) are intervals of exactly 60x1000=60,000 milliseconds
+each.
+
+*hours (h)* +
+All hours begin at 00 minutes and 00 seconds. +
+* One hour (1h) is the interval between 00:00 minutes of the first hour and 00:00
+minutes of the following hour in the specified timezone, compensating for any
+intervening leap seconds, so that the number of minutes and seconds past the hour is the same at the start and end.
+* Multiple hours (_n_h) are intervals of exactly 60x60x1000=3,600,000 milliseconds each.
+
+*days (d)* +
+All days begin at the earliest possible time, which is usually 00:00:00
+(midnight).
+* One day (1d) is the interval between 00:00:00 of the first day and 00:00:00
+of the following day in the specified timezone, compensating for any intervening
+time changes, so that the number of hours, minutes, and seconds of the day is
+the same at the start and end.
+* Multiple days (_n_d) are intervals of exactly 24x60x60x1000=86,400,000 milliseconds each.
+
+*weeks (w)* +
+* One week (1w) is the interval between the start day_of_week:hour:minute:second
+and the same day and time of the following week in the specified timezone, so that
+the day of the week and time of day are the same at the start and end.
+* Multiple weeks (_n_w) are not supported.
+
+*months (m)* +
+* One month (1m) is the interval between the start day of the month and time of
+day and the same day of the month and time of the following month in the specified
+timezone, so that the day of the month and time of day are the same at the start
+and end.
+* Multiple months (_n_m) are not supported.
+
+*quarters (q)* +
+* One quarter (1q) is the interval between the start day of the month and time
+time of day and the same day of the month and time of day three months later,
+so that the day of the month and time of day are the same at the start and end.
+* Multiple quarters (_n_q) are not supported.
+
+*years (y)* +
+* One year (1y) is the interval between the start day of the month and time of 
+day and the same day of the month and time of day the following year in the 
+specified timezone, so that the date and time are the same at the start and end.
+* Multiple years (_n_y) are not supported.
+
+Widely distributed applications must also consider vagaries such as countries that
+start and stop daylight savings time at 12:01 A.M., so end up with one minute of 
+Sunday followed by an additional 59 minutes of Saturday every fall, and countries
+that decide at whim to move across the international date line. Situations like
+that can make irregular timezone offsets seem easy. 
+
+As always, rigorous testing will ensure that your time interval specification is
+what you intend it to be.
+
+WARNING:
+To avoid unexpected results, all connected servers and clients must sync to the
+same network time service.
```

Comment:
I think we should leave out the "same network time service" bit. Syncing to any (non-broken) time service is going to give you reasonably accurate clock sync. For instance, by default, Windows machines use a different default time service from Linux machines, and this is fine. How about:

> To avoid unexpected results, ensure that the clocks on all servers and clients are well-synchronized using a network time service or similar functionality.

Reply:
See issue #29753

Reply:
Sure, that doesn't mean it's right to say "the same network time service". Syncing to any (non-broken) time service would do, as they are all sufficiently in sync.

Reply:
Changed "the same" to "a reliable"